server/src/inmantals/server.py

```diff
@@ -76,110 +79,145 @@ class Folder:
+
+        self.folder_uri = os.path.abspath(folder_uri.path)
         self.handler = handler  # Keep a reference to the handler for cleanup
+        self.kind: object
 
         # Check that we are working inside an existing project:
-        project_file: str = os.path.join(folder_uri.path, module.Project.PROJECT_FILE)
+        project_file: str = os.path.join(self.folder_uri, module.Project.PROJECT_FILE)
         if os.path.exists(project_file):
             self.inmanta_project_dir = os.path.dirname(project_file)
+            self.kind = module.Project
         else:
-            self.inmanta_project_dir = None
-
-    def get_project_dir(self) -> Optional[str]:
-        """
-        If this folder holds an inmanta project, this returns the path to this project.
-        If this folder holds an inmanta module, this returns the path of the temporary
-        inmanta project used to load this module.
-        """
-        if not self.inmanta_project_dir:
-            return None
-        if isinstance(self.inmanta_project_dir, str):
-            return self.inmanta_project_dir
-        else:
-            return self.inmanta_project_dir.name
+            self.inmanta_project_dir = self.create_tmp_project()
 
     def get_setting(self, key: str, default: Optional[str] = None) -> Optional[str]:
         """
         Get an inmanta extension setting for this folder
         """
         if key == "compiler_venv_path":  # support for versions of core that require a compiler venv.
-            return os.path.join(self.get_folder_path(), ".env-ls-compiler")
+            return os.path.join(self.folder_uri, ".env-ls-compiler")
         try:
             # Look for some custom settings for this folder
-            with open(os.path.join(self.get_folder_path(), ".vscode", "settings.json")) as settings_file:
+            with open(os.path.join(self.folder_uri, ".vscode", "settings.json")) as settings_file:
                 settings: dict[str, str] = json.load(settings_file)
                 return settings.get(key, default)
         except FileNotFoundError:
             return default
 
-    @classmethod
-    def unpack_workspaces(cls, workspace_folders: Sequence[object], ls_handler: "InmantaLSHandler") -> Dict[str, "Folder"]:
-        return {folder["uri"]: Folder(URI(folder["uri"]), folder["name"], ls_handler) for folder in workspace_folders}
-
     def cleanup(self) -> None:
-        logger.info(f"calling cleanup for {self}")
+        logger.info("Cleaning up %s...", self)
         if self.inmanta_project_dir:
-            self.handler.remove_folder(str(self.folder_uri))
             self.inmanta_project_dir.cleanup()
 
-    def get_folder_path(self) -> str:
-        """
-        Convert the folders's uri into a regular path
-        (https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#uri)
-        """
-        return self.folder_uri.path
+    def install_v2_module_editable_mode(self):
```

Comment:
Could you add a docstring that declares the precondition: this method assumes that the project is already set up (`module.Project.set()`), including the module repos config.